python/revng/support/artifacts.py

```diff
@@ -189,11 +190,21 @@ class LLVMArtifact(Artifact):
 
     def module(self, name: str = "module"):
         llvmcpy = get_llvmcpy()
+        context = llvmcpy.get_global_context()
         buffer = llvmcpy.create_memory_buffer_with_memory_range_copy(
             self._data, len(self._data), name
         )
-        context = llvmcpy.get_global_context()
-        return context.parse_ir(buffer)
+        try:
+            return context.parse_ir(buffer)
+        except llvmcpy.LLVMException as e:
+            if self._data[:4] != zstd.FRAME_HEADER:
+                raise e
```

Comment:
Can't we preemptively check this instead of doing the trial and error thing?

Reply:
The check is backwards: it's unknown whether the llvm version is ours which is patched to read `zstd`s or not, the only way is to try to read it and then do something if an exception is thrown

Reply:
My idea was no to rely on LLVM supporting this at all.

This library is primarily designed to be used *out of* orchestra.

Reply:
Sure, but if we *happen* to be within orchestra we save having to call zstd.
If you prefer we do this regardless, then that's fine

Reply:
I don't think we'll ever use this within orchestra.

But in any case, we're just decompressing in Python (but still using C code, I assume) as opposed to do the same within our LLVM fork. I don't think there's a performance penalty. If we can avoid dealing with exceptions, let's do it. There are many reasons why there could be a `llvmcpy.LLVMException`. Here we are suppressing and trying to decompress it. If the reason for the failure was *not* due to the compression (it was not), we're hiding the error and reporting the zstd failure to decompress (it wasn't compressed).